File: server/v2.golang/router/routes/index/monitor.go
package index

import (
	types "kooperlingohr/portfolio/Types"
	"kooperlingohr/portfolio/controllers/database"
	"kooperlingohr/portfolio/router/middleware/responses"
	"kooperlingohr/portfolio/utils"
	"net/http"
	"time"
)

func Monitor(w http.ResponseWriter, r *http.Request) {
	if r.Method == http.MethodPost {
		var body types.MonitorPayload
		datetime := time.Now().Format("2006-01-02 15:04:05.000000")
		utils.ParseReqParams(r.URL.Query(), &body)
		headerColumns := []interface{}{nil, datetime}

		insertStatement := "INSERT INTO monitor VALUES (?, ?, ?, ?, ?);"
		data := append(headerColumns, body.Uuid, body.SessionID, body.Page)
		database.Insert(insertStatement, data)

		if body.PrevPage != "NULL" {
			insertStatement = "INSERT INTO route_track VALUES (?, ?, ?, ?, ?);"
			data = append(headerColumns, body.SessionID, body.PrevPage, body.Page)
			database.Insert(insertStatement, data)
		}
		responses.BuildPlainSuccess(w, 200)

	} else {
		// 405
	}
}

File: server/v3.rust/src/routes/index/monitor.rs
use rocket::{get, post, http::Status, serde::json::Json, request};
use serde::Deserialize;
use std::{fs};
use sqlx::{migrate::MigrateDatabase, Row, Sqlite, SqlitePool};
#[path = "../../types/response.rs"] mod response;

#[derive(Deserialize)]
pub struct MonitorPayload {
	session_id: String, 
	uuid: String,
	page: String,
	prevPage: Option<String>,
}

#[post("/monitor", data="<input>")]                                                                            
pub async fn monitorRoutePost(input: Json<MonitorPayload>) -> Result<Json<response::GenericResponse<String>>, Status> {
    const DB_URL: &str = "sqlite://server/data/database.db";
    let db = SqlitePool::connect(DB_URL).await.unwrap();
    let time = format!("{}", chrono::Local::now().format("%Y-%m-%d %H:%M:%S"));
	let insertStatementMonitor = "INSERT INTO monitor VALUES (NULL, ?, ?, ?, ?);";
    let result_monitor = sqlx::query(insertStatementMonitor)
        .bind(&time)
        .bind(&input.uuid)
        .bind(&input.session_id)
        .bind(&input.page)
    .execute(&db)
    .await
    .unwrap();

	if let Some(e) = &input.prevPage {
		let insertStatementTrack = "INSERT INTO route_track VALUES (NULL, ?, ?, ?, ?);";
        let result_track = sqlx::query(insertStatementTrack).bind(&time).bind(&input.session_id).bind(&input.prevPage).bind(&input.page)
        .execute(&db)
        .await
        .unwrap();
	}
    Ok(Json(response::GenericResponse {
        success: true,
        data: None,
        errorMessage: None,
    }))
}
